Shared/Fun_GenApp.swift

```diff
@@ -6,12 +6,28 @@
 //
 
 import SwiftUI
+import UIKit
+import Firebase
 
 @main
 struct Fun_GenApp: App {
+    // firebase init
+    @UIApplicationDelegateAdaptor(AppDelegate.self) var appDelegate
+    
     var body: some Scene {
         WindowGroup {
+            let user = UserViewModel()
             ContentView()
+                .environmentObject(user)
         }
     }
 }
+
+// firebase init
+class AppDelegate: NSObject, UIApplicationDelegate {
+  func application(_ application: UIApplication,
+                   didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    FirebaseApp.configure()
+    return true
+  }
```

Comment:
1. Please use 4-space indentation
2. When the parameter list is very long like this, you can begin the parameter list on a separate line:

```suggestion
    func application(
        _ application: UIApplication,
        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
    ) -> Bool {
        FirebaseApp.configure()
        return true
    }
```

Reply:
Fixed

Reply:
The indentation still looks a bit off, can you try to use <kbd>control</kbd> + <kbd>i</kbd> (or menu optoins Editor > Structure > Re-indent) to automatically format this section in Xcode?

Reply:
Fixed